### DISCUSSION TITLE
Not enough bytes read from file

### DISCUSSION THREAD

**Discussion by sonic-purebase** [appy-one/acebase#252] [Q&A] [answered]
I previously was using AceBase successfully, distributed the code to another person, and after some months they alerted me it wasn't working. I ran the code again on my machine, and it no longer works for me either. The problem section is below. AceBase version is 1.29.5. The error I get is:

```
Error: Not enough bytes read from file at index 131072, expected 128 but got 0 at NodeReader.readHeader
```
It is caused by the line at ```const ref = await AceServer.ref('event').set({```

if the process is left to run long enough, I eventually get this:
```
[testinghere] write lock on path "/event" by tid lu6f4lcr0001j0i5ec9bqgtg (_updateNode) is taking a long time to complete [1]
[testinghere] lock :: write lock on path "/event" by tid lu6f2nwu0002q2i54s6rh5ty (_updateNode) took too long
```

It does log ```'server ready'```, so the AceServer is initialized. Changing the function from ```set()``` to ```update()``` results in the same error. I have tried clearing the node environment and re-installing, with no change.

```code
  const dbPath = args['dbPath']
  var dbDir = dbPath.split(path.sep).pop() // get last element in split of filepath, which will be the name of the last folder
  dbDir = dbDir.replace('.acebase', "") // remove acebase from name
  const parentDir = path.parse(dbPath)['dir'] // get path to parent directory of acebase folder

  try {
    AceServer.close() //close existing db
    AceServer = new AceBase(dbDir, {storage: { path: parentDir }}) // start a new db
  } catch (error) {
    AceServer = new AceBase(dbDir, {storage: { path: parentDir }}) // start a new db
  }

  AceServer.ready(() => {
    console.log('server ready')
  })
  try {
    const ref = await AceServer.ref('event').set({
      name: args['eventName'],
      _id: args['eventID']
    })
  }
```

**Comment by appy-one** [comment]
This looks like the database file has been corrupted, which is probably caused by multiple processes writing to the same file without using IPC (inter process communication). Is your code running in a cluster or a single instance?

**Reply by sonic-purebase** [reply, marked as answer]
Single instance. It looks like the issue was calling `.close()`, which doesn't exist. Looking back, I don't see anything in the documentation about it, so I'm not sure why I was doing that. Thank you, I think this is solved.
